### PR TITLE
Add vof phase factor in heat transfer laser source and radiation term at the free surface

### PR DESCRIPTION
**Pull request by hepap**
# Description of the problem

For additive manufacturing simulations the both the heat transfer and VOF solvers, there were two required changes:
1. one issue to be addressed: the metal AND the air were heated by the  laser instead of only the metal. 
2. one feature to be added, that is the radiation cooling at the metal/air interface, which is a significant cooling mechanism in additive manufacturing simulations.

Both changes were implemented in the heat transfer solver:
1.  For the air "permeability" to the laser, the class `HeatTransferAssemblerLaserVOF` was implemented. The laser source term is computed as in the previously implemented class `HeatTransferAssemblerLaser`, however, it is multiply by the phase fraction (computed in the VOF solver) to activate the laser source term in the metal (phase fraction = 1) and deactivate it in the air (phase fraction = 0).
2. For the  radiation cooling, the class `HeatTransferAssemblerFreeSurfaceRadiationVOF` was implemented. Also, the parameters struct  `Laser_FreeSurfaceRadiation` was implemented to described the required parameters for the radiation term and was added as an attribute of the parameters class `Laser`.  The radiation term is applied only in the cells at the air/metal interface. It is treated as a source term instead of a boundary term using the phase fraction gradient as the interface indicator.
3. Both features are enabled in the `HeatTransfer` class.

# Description of the solution

- The air is now "permeable" to the laser heating source, i.e., it is not heated by the laser.
- The radiation term was added (following the Stefan-Boltzmann law of radiation) and verified with a steady-state analytical solution. 

# How Has This Been Tested?

- [applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.prm and .out] An application test with an analytical solution was added for the radiation term. The computational domain is a square of an unit side length: the bottom half is fluid 1 (metal) and the top half is fluid 0 (air). The radiation term is applied at y=0.5. 
- all the other applications tests passed.

# Documentation

- [doc/source/parameters/cfd/laser_heat_source.rst ] The added parameters for the radiation term were documented.

# Future changes

- An example will be added.

### CODE REVIEW

File: applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.prm
Comment (blaisb) [inline applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.prm:18]: Can you indent the prm file using the prmindent utility in /contrib/utilities?

File: doc/source/parameters/cfd/laser_heat_source.rst
Comment (blaisb) [inline doc/source/parameters/cfd/laser_heat_source.rst:54]: Nice documentation. Good job @hepap  :)

File: source/core/parameters.cc
Comment (blaisb) [inline source/core/parameters.cc:885]: you can use exponential notation
I suggest writing it as 5.6703e-8 instead of putting tha gazillion zeroes. It's easier the my brain :).

File: source/solvers/heat_transfer.cc
Comment (blaisb) [inline source/solvers/heat_transfer.cc:291]: Chill design. Good job.

**Review by blaisb — APPROVED** [review]
I don't really have any comments. Amazing work.
Please wait for a second review, then i'll merge.

File: source/core/parameters.cc
Comment (hepap) [reply source/core/parameters.cc]: Done :)

File: applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.prm
Comment (hepap) [reply applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.prm]: Done :)

File: doc/source/parameters/cfd/laser_heat_source.rst
Comment (oguevremont) [inline doc/source/parameters/cfd/laser_heat_source.rst:27]:
```suggestion
          set Stefan-Boltzmann constant = 5.6703e-8
```

File: include/solvers/heat_transfer_assemblers.h
Comment (oguevremont) [inline include/solvers/heat_transfer_assemblers.h:348]:
```suggestion
```

File: include/solvers/heat_transfer_assemblers.h
Comment (oguevremont) [inline include/solvers/heat_transfer_assemblers.h:398]:
```suggestion
```

**Review by oguevremont — APPROVED** [review]
Excellent work!

File: applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.output
Comment (hepap) [reply applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.output:17]: Are you talking about the 0.000000e+00? or the 1.768316e-02?

Comment (oguevremont) [reply applications_tests/gls_navier_stokes_2d/heat_transfer_vof_radiation.output:17]: I suspect 0.000000e+00 will stay the same all the time, so for 1.768316e-02

**Pull request merged**
